pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -438,7 +438,20 @@ func (nc *Controller) doNoScheduleTaintingPass(node *v1.Node) error {
 			}
 		}
 	}
+	if node.Spec.Unschedulable {
+		// If unscheduable, append related taint.
```

Comment:
what removes this taint when the node becomes schedulable again?

Reply:
definitely needs a test demonstrating this lifecycle works properly

Reply:
Good point. This might break `kubectl cordon/uncordon` too.

Reply:
@janetkuo +1 as we need to ensure how the taint is removed when `kubectl uncordon`

Reply:
The removal is handled here: https://github.com/kubernetes/kubernetes/pull/61161/files#diff-43035107687eb30550696751ac1066e4R458 . we'll get target taints according to the conditions, and the exist taints on the node; and then get `taintsToAdd` and `taintsToDel` by `TaintSetDiff`. `SwapNodeControllerTaint` will update the taints accordingly.

re `a test`: definitely :)

Reply:
added integration test for unschedulable; tested removal manually.